packages/cardpay-reward-programs/tests/rules/test_min_spend.py

```diff
@@ -50,58 +40,48 @@ def rule(request):
 
 class TestMinSpendSingle:
     @pytest.mark.parametrize(
-        "rule,ans",
+        "rule,summary",
         zip(
             itertools.product(payment_cycle_length_ls, min_spend_ls),
-            ans_ls,
+            summaries,
         ),
         indirect=["rule"],
     )
-    def test_run(self, rule, ans, indexed_data):
-        df_hash, summary = ans
+    def test_run(self, rule, summary, indexed_data):
         start_block = 24000000
         end_block = 26000000
-        df = rule.run(start_block, end_block)
-        payment_list = rule.df_to_payment_list(df)
+        rule.payment_cycle_length = end_block - start_block
+        payment_list = rule.run(end_block, "0x0")
         computed_summary = rule.get_summary(payment_list)
         assert computed_summary["total_reward"][0] == summary["total_reward"]
         assert computed_summary["unique_payee"][0] == summary["unique_payee"]
 
 
-multiple_df_hashes = [
-    "016f5967731732f8879315395291cd9c95ad83e5c0364b11e41b0efa3980ace4",
-    "ab33d395cbf0ebb1d5baa21ae2d3b0af5093ddef6e5c0d93f42a5a7ca1ecb08d",
-    "c3f4c2d3c1978f2d1770e2d1c564951ff6ff028d2d68b2b442d77cf41351edf6",
-]
+
 multiple_summaries = [
```

Comment:
It follows here. That the reason these test values changed is because of the way `aggregate` works. 

My way: Combine dfs in consideration of the `min_spend` criteria
Your way: Combine payment list

Again lets remove this. Sorry if it was not clearly communicated.